pkg/bot/slack.go

```diff
@@ -289,23 +290,24 @@ func (b *Slack) SendEvent(ctx context.Context, event events.Event) error {
 	return errs.ErrorOrNil()
 }
 
-func (b *Slack) getChannelsToNotify(event events.Event) []string {
+func (b *Slack) getChannelsToNotify(event events.Event, eventSources []string) []string {
 	// support custom event routing
 	if event.Channel != "" {
 		return []string{event.Channel}
 	}
 
-	// TODO(https://github.com/kubeshop/botkube/issues/596): Support source bindings - filter events here or at source level and pass it every time via event property?
-	var channelsToNotify []string
-	for _, channelCfg := range b.getChannels() {
-		if !channelCfg.notify {
-			b.log.Info("Skipping notification for channel %q as notifications are disabled.", channelCfg.Identifier())
-			continue
+	var out []string
+	for _, cfg := range b.getChannels() {
+		switch {
+		case !cfg.notify:
+			b.log.Info("Skipping notification for channel %q as notifications are disabled.", cfg.Identifier())
+		default:
+			if utils.Intersect(eventSources, cfg.Bindings.Sources) {
+				out = append(out, cfg.Identifier())
+			}
```

Comment:
I think that's easier to read - and we have less indentations:
```suggestion
		if !cfg.notify {
			b.log.Info("Skipping notification for channel %q as notifications are disabled.", cfg.Identifier())
			continue
		}

		if utils.Intersect(eventSources, cfg.Bindings.Sources) {
			out = append(out, cfg.Identifier())
		}
```

Reply:
Marking this comment as unresolved, as I don't see a place where it was applied 🤔 This is a minor comment, but consider simplifying the code as suggested in all bots 🙂

Reply:
Will have a look. Thanks for the heads up.

Reply:
It's a matter of style. I'm happy with guard clauses. Updated.
(running tests locally to double check all is well)

Reply:
Thanks for applying it for Slack. I think it looks better as there's less indentations now.

> consider simplifying the code as suggested **in all bots** 🙂

Please consider to apply it for all other places 🙂